test/common/buffer/watermark_buffer_test.cc

```diff
@@ -486,12 +481,9 @@ TEST_F(WatermarkBufferTest, OverflowWatermarkDisabledOnVeryHighValue) {
 
   // Reserve and commit additional space on the buffer beyond the expected
   // high_watermark_threshold * overflow_multiplier threshold.
-  // Adding high_watermark_threshold * overflow_multiplier - buffer1.length() + 1 bytes
-  Buffer::RawSlice iovecs[2];
-  uint64_t num_reserved = buffer1.reserve(
-      high_watermark_threshold * overflow_multiplier - buffer1.length() + 1, iovecs, 2);
-  EXPECT_GE(num_reserved, 1);
-  buffer1.commit(iovecs, num_reserved);
+  const uint64_t size = high_watermark_threshold * overflow_multiplier - buffer1.length() + 1;
+  auto reservation = buffer1.reserveApproximately(size);
```

Comment:
Please use reserveSingleSlice.

It is unclear from text context that size is guaranteed to be less than 128kb.